stickynotes/stickynotes_applet.c

```diff
@@ -31,595 +31,638 @@ StickyNotes *stickynotes = NULL;
+        GTK_WIDGET (&GTK_CHECK_BUTTON (gtk_builder_get_object (stickynotes->builder,
+                                                               "sys_font_check"))->toggle_button);
+    stickynotes->w_prefs_sticky =
+        GTK_WIDGET (&GTK_CHECK_BUTTON (gtk_builder_get_object (stickynotes->builder,
+                                                               "sticky_check"))->toggle_button);
+    stickynotes->w_prefs_force =
+        GTK_WIDGET (&GTK_CHECK_BUTTON (gtk_builder_get_object (stickynotes->builder,
+                                                               "force_default_check"))->toggle_button);
+    stickynotes->w_prefs_desktop =
+        GTK_WIDGET (&GTK_CHECK_BUTTON (gtk_builder_get_object (stickynotes->builder,
+                                                               "desktop_hide_check"))->toggle_button);
+
+    g_signal_connect (G_OBJECT (stickynotes->w_prefs),
+                      "response",
+                      G_CALLBACK (preferences_response_cb), NULL);
+    g_signal_connect (G_OBJECT (stickynotes->w_prefs),
+                      "delete-event",
+                      G_CALLBACK (preferences_delete_cb), NULL);
+    g_signal_connect_swapped (G_OBJECT (stickynotes->w_prefs_width),
+                              "value-changed",
+                              G_CALLBACK (preferences_save_cb), NULL);
+    g_signal_connect_swapped (G_OBJECT (stickynotes->w_prefs_height),
+                              "value-changed",
+                              G_CALLBACK (preferences_save_cb), NULL);
+    g_signal_connect_swapped (G_OBJECT (stickynotes->w_prefs_sys_color),
+                              "toggled",
+                              G_CALLBACK (preferences_save_cb), NULL);
+    g_signal_connect_swapped (G_OBJECT (stickynotes->w_prefs_sys_font),
+                              "toggled",
+                              G_CALLBACK (preferences_save_cb), NULL);
+    g_signal_connect (G_OBJECT (stickynotes->w_prefs_color),
+                      "color-set",
+                      G_CALLBACK (preferences_color_cb), NULL);
+    g_signal_connect (G_OBJECT (stickynotes->w_prefs_font_color),
+                      "color-set",
+                      G_CALLBACK (preferences_color_cb), NULL);
+    g_signal_connect (G_OBJECT (stickynotes->w_prefs_font),
+                      "font-set",
+                      G_CALLBACK (preferences_font_cb), NULL);
+    g_signal_connect_swapped (G_OBJECT (stickynotes->w_prefs_sticky),
+                              "toggled",
+                              G_CALLBACK (preferences_save_cb), NULL);
+    g_signal_connect_swapped (G_OBJECT (stickynotes->w_prefs_force),
+                              "toggled",
+                              G_CALLBACK (preferences_save_cb), NULL);
+    g_signal_connect_swapped (G_OBJECT (stickynotes->w_prefs_desktop),
+                              "toggled",
+                              G_CALLBACK (preferences_save_cb), NULL);
+
```

Comment:
no need to use G_OBJECT cast for 1st argument of g_signal_connect_swapped / g_signal_connect

Reply:
You mean changing code to...., for example?
```
g_signal_connect_swapped (stickynotes->w_prefs_desktop, "toggled",
                          G_CALLBACK (preferences_save_cb), NULL);
```

Reply:
Yes

```
As you can see below in the class hierarchy, all GtkWidgets are derived from the
GObject base class. This means you can use a widget in any place the function
asks for an object. Note that many functions accept a gpointer so there is no
need to cast the type.

For example:

g_signal_connect( button, "clicked",
                   G_CALLBACK (callback_function), callback_data);

This casts the button into an object, and provides a cast for the function pointer
to the callback.

```


https://developer.gnome.org/gtk-tutorial/stable/c450.html

Reply:
Done.